crates/bevy_asset/src/io/metadata.rs

```diff
@@ -0,0 +1,58 @@
+/// A enum representing a type of file.
+#[derive(Copy, Clone, Debug, PartialEq)]
```

Comment:
```suggestion
#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
```
Also derive `Eq` and `Hash`